src/main/java/encriptionAlgorithms/basicAlgorithms/ShiftMultiplyEncryption.java

```diff
@@ -5,8 +5,22 @@
 
 import java.math.BigInteger;
 
+//singleton class
 public class ShiftMultiplyEncryption extends BasicEncryption {
 
+    static ShiftMultiplyEncryption instanceShiftMultiplyEncryption = null;
+    private ShiftMultiplyEncryption() {
+    }
+
+    public static ShiftMultiplyEncryption CreateShiftMultiplyEncryption() {
```

Comment:
```suggestion
    public static ShiftMultiplyEncryption getShiftMultiplyEncryption() {
```
Will return the alg. sometimes it will create him

Reply:
use getInstance
see https://www.geeksforgeeks.org/singleton-class-java/

Reply:
done